backend/organization/views/organization_views.py

```diff
@@ -21,3 +30,126 @@ def get_serializer_class(self):
+                }, status=status.HTTP_400_BAD_REQUEST)
+
+        organization, created = Organization.objects.get_or_create(name=request.data['name'])
+
+        if created:
+            organization.url_slug = organization.name.replace(" ", "") + str(organization.id)
+
+            if 'image' in request.data:
+                organization.image = request.data['image']
+            if 'background_image' in request.data:
+                organization.background_image = request.data['background_image']
+
+            if 'parent_organization' in request.data:
+                try:
+                    parent_org = Organization.objects.get(id=int(request.data['parent_organization']))
+                except Organization.DoesNotExist:
+                    return Response({
+                        'message': 'Parent organization not found.'
+                    }, status=status.HTTP_404_NOT_FOUND)
+
+                organization.parent_organization = parent_org
+
+            if 'country' in request.data:
+                organization.country = request.data['country']
+            if 'state' in request.data:
+                organization.state = request.data['state']
+            if 'city' in request.data:
+                organization.city = request.data['city']
+            if 'short_description' in request.data:
+                organization.short_description = request.data['short_description']
+            organization.save()
+
+            roles = Role.objects.all()
+            for member in request.data['team_members']:
+                user_role = roles.filter(id=int(member['permission_type_id'])).first()
+                try:
+                    user = User.objects.get(id=int(member['user_id']))
+                except User.DoesNotExist:
+                    logger.error("Passed user id {} does not exists".format(member['user_id']))
+                    continue
+
+                if user:
+                    OrganizationMember.objects.create(
+                        user=user, organization=organization, role=user_role
+                    )
+                    logger.info("Organization member created {}".format(user.id))
+
+            return Response({
+                'message': 'Organization {} successfully created'.format(organization.name)
+            }, status=status.HTTP_201_CREATED)
+        else:
+            return Response({
+                'message': 'Organization with name {} already exists'.format(request.data['name'])
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+
+class OrganizationAPIView(RetrieveUpdateDestroyAPIView):
+    permission_classes = [OrganizationReadWritePermission]
+    serializer_class = OrganizationSerializer
+    lookup_field = 'url_slug'
+
+    def get_queryset(self):
+        return Organization.objects.filter(url_slug=str(self.kwargs['url_slug']))
+
+    def perform_update(self, serializer):
+        serializer.save()
+        return serializer
+
+    def perform_destroy(self, instance):
```

Comment:
For destroy I think you should definitely have 'ALL' permission. (Only the owner should be able to delete an organization)

Reply:
Good point. Will do that through permissions.

Reply:
This has been resolved.